### PR TITLE
[BUGFIX] Restricted pages are not being indexed in Typo3 10 

### PR DESCRIPTION
**Pull request by dkd-kaehm**
Fixes: #2634

### CODE REVIEW

**Comment by dkd-kaehm** [comment]
@georgringer 
Could you please tests this PR within TYPO3 9.5 and 10.4?

File: Classes/Middleware/FrontendUserAuthenticator.php [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        if (Util::getIsTYPO3VersionBelow10()
+            || !$request->hasHeader(PageIndexerRequest::SOLR_INDEX_HEADER)
+        ) {
+            return $handler->handle($request);
+        }
+
+        // disable TSFE cache for TYPO3 v10
+        $request = $request->withAttribute('noCache', true);
```
Comment (georgringer) [inline Classes/Middleware/FrontendUserAuthenticator.php:86]: this won't work if no_cache is disabled via `LocalConfiguration.php`
I propose disabling the cache with a middleware/hook when indexing, something like https://github.com/studiomitte/powermail_freebie/blob/master/Classes/Hooks/CacheHook.php

File: Classes/IndexQueue/FrontendHelper/PageIndexer.php [hunk 2]
```diff
@@ -69,33 +74,29 @@ class PageIndexer extends AbstractFrontendHelper implements SingletonInterface
     protected $responseData = [];
 
     /**
-     * @var \TYPO3\CMS\Core\Log\Logger
+     * @var Logger
```
Comment (georgringer) [inline Classes/IndexQueue/FrontendHelper/PageIndexer.php:77]: it makes reviewing really hard when you mix such CGL issues with bugfixing

File: Classes/Middleware/FrontendUserAuthenticator.php [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        return $request;
+    }
+
+    /**
+     * Gets the access rootline as defined by the request.
+     *
+     * @param PageIndexerRequestHandler $handler
+     * @return Rootline The access rootline to use for indexing.
+     */
+    protected function getAccessRootline(PageIndexerRequestHandler $handler): ?Rootline
```
Comment (georgringer) [inline Classes/Middleware/FrontendUserAuthenticator.php:146]:
```suggestion
    protected function getAccessRootline(PageIndexerRequestHandler $handler): Rootline
```

shouldn't that always return a Rootline object

File: Classes/Middleware/FrontendUserAuthenticator.php (hunk 1 above)
Comment (dkd-kaehm) [reply Classes/Middleware/FrontendUserAuthenticator.php:86]: I'll create a new issue for that. We should clean up the old stuff anyway, there are multiple things, which try to disable caches.

**Comment by georgringer** [comment]
@dkd-kaehm do you want to merge that?

**Comment by dkd-kaehm** [comment]
Yes, incl. Backport onto 11.

**Comment by dkd-kaehm** [comment]
@georgringer 
Do you have some pages with restrictions to the content elements? Does this patch work with them?

**Pull request merged**